core/state/caching_versa_db.go

```diff
@@ -0,0 +1,297 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+
+	versa "github.com/bnb-chain/versioned-state-database"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/lru"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/ethereum/go-ethereum/trie/trienode"
+	"github.com/ethereum/go-ethereum/triedb"
+)
+
+type cachingVersaDB struct {
+	triedb        *triedb.Database
+	versionDB     versa.Database
+	codeDB        ethdb.KeyValueStore
+	codeSizeCache *lru.Cache[common.Hash, int]
+	codeCache     *lru.SizeConstrainedCache[common.Hash, []byte]
+
+	accTree  *VersaTree
```

Comment:
Where is VersaTree impl? And why to have a "accTree"?

Reply:
Facilitate OpenStorageTree to avoid reopen.